File: languages/ruby/lib/oso/oso/http.rb
# frozen_string_literal: true

module Oso
  module Oso
    # An HTTP resource.
    class Http
      def initialize(hostname: nil, path: nil, query: nil)
        @hostname = hostname
        @path = path
        @query = query
      end

      private

      attr_reader :hostname, :path, :query
    end
  end
end

File: languages/js/src/Http.ts
export class Http {
  readonly hostname: string;
  readonly path: string;
  readonly query: Map<string, string>;

  constructor(hostname: string, path: string, query: Map<string, string>) {
    this.hostname = hostname;
    this.path = path;
    this.query = query;
  }
}
